logstash-core/src/main/java/co/elastic/logstash/api/Input.java

```diff
@@ -0,0 +1,110 @@
+package co.elastic.logstash.api;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Scanner;
+import java.util.concurrent.CountDownLatch;
+
+import org.logstash.execution.queue.QueueWriter;
+
+/**
+ * A Logstash Pipeline Input pushes to a {@link QueueWriter}.
+ */
+public interface Input extends Plugin {
+
+    /**
+     * Start pushing {@link org.logstash.Event} to given {@link QueueWriter}.
+     * @param writer Queue Writer to Push to
+     */
+    void start(QueueWriter writer);
+
+    /**
+     * Stop the input.
+     * Stopping happens asynchronously, use {@link #awaitStop()} to make sure that the input has
+     * finished.
+     */
+    void stop();
+
+    /**
+     * Blocks until the input execution has finished.
+     * @throws InterruptedException On Interrupt
+     */
+    void awaitStop() throws InterruptedException;
+
+    @LogstashPlugin(name = "java-one-input-event")
+    final class OneInputEvent implements Input {
+
+        public OneInputEvent(final Configuration configuration, final Context context) {
+            // do nothing
+        }
+
+        @Override
+        public void start(QueueWriter writer) {
+            writer.push(Collections.singletonMap("oneEvent", "isDone"));
+        }
+
+        @Override
+        public void stop() {
+
+        }
+
+        @Override
+        public void awaitStop() throws InterruptedException {
+
+        }
+
+        @Override
+        public Collection<PluginConfigSpec<?>> configSchema() {
+            return Collections.EMPTY_LIST;
+        }
+    }
+
+    @LogstashPlugin(name = "stream")
+    final class StreamInput implements Input {
```

Comment:
Definitely. These are examples from the really early days of the Java API and I'll probably just remove them since the example Java plugins in https://github.com/logstash-plugins should take their place.